### PR TITLE
[`tryceratops`] Add `BaseException` to raise-vanilla-class rule (`TRY002`)

### LINKED ISSUE #12612: `TRY002` does not take into account `BaseException`
Spotted in https://github.com/home-assistant/core/pull/123021

It seems that pylint doesn't like Exception nor BaseException
https://pylint.pycqa.org/en/latest/user_guide/messages/warning/broad-exception-raised.html

But ruff is quite happy to let BaseException pass

I think both should trigger a warning, same as regular Exception

It should be an easy fix in https://github.com/astral-sh/ruff/blob/a3e67abf4ce7b519152c03ff441424c18e4c18ee/crates/ruff_linter/src/rules/tryceratops/rules/raise_vanilla_class.rs#L71

### PR DESCRIPTION
**Pull request by epenet**
Fixes #12612

### CODE REVIEW

**Comment by MichaReiser** [comment]
Code looks good to me. We only have to decide on https://github.com/astral-sh/ruff/issues/12612#issuecomment-2263942167

**Comment by AlexWaygood** [comment]
This makes sense to me. It's unfortunate to deviate from `tryceratops` here, but I think it definitely makes sense to include `BaseException` as well as `Exception`.

Should we also include `ExceptionGroup` and `BaseExceptionGroup`, though? (That's a genuine question that I'm not sure the answer to -- I haven't had much opportunity to use these fancy new features yet! It might be that they belong in this rule, or it might be that they deserve their own rule... not sure.)

File: crates/ruff_linter/src/rules/tryceratops/rules/raise_vanilla_class.rs [hunk 1]
```diff
@@ -68,7 +68,12 @@ pub(crate) fn raise_vanilla_class(checker: &mut Checker, expr: &Expr) {
     } else {
         expr
     };
-    if checker.semantic().match_builtin_expr(node, "Exception") {
+
+    let semantic = checker.semantic();
+
+    if semantic.match_builtin_expr(node, "Exception")
+        || semantic.match_builtin_expr(node, "BaseException")
```
Comment (AlexWaygood) [inline crates/ruff_linter/src/rules/tryceratops/rules/raise_vanilla_class.rs:75]: This is resolving the qualified name twice when it only needs to resolve it once -- we can micro-optimize this a little bit:

```suggestion
    if checker
        .semantic()
        .resolve_qualified_name(node)
        .is_some_and(|qualified_name| {
            matches!(qualified_name.segments(), ["" | "builtins", "Exception" | "BaseException"])
        })
```

**Comment by AlexWaygood** [comment]
Let's land this for now, we can tackle the `ExceptionGroup` and `BaseExceptionGroup` issues as a followup. I think this is a strict improvement.

Thanks @epenet!

**Pull request merged**